src/Views/OperatingSystemView.vala

```diff
@@ -713,4 +731,125 @@ public class About.OperatingSystemView : Gtk.Box {
             add_css_class ("link");
         }
     }
+
+    private class SponsorUsRow : Gtk.ListBoxRow {
+        public string uri { get; construct; }
+
+        private Gtk.Revealer details_revealer;
+
+        public SponsorUsRow (string uri) {
+            Object (
+                uri: uri
+            );
+        }
+
+        class construct {
+            set_accessible_role (LINK);
+        }
+
+        construct {
+            var image = new Gtk.Image.from_icon_name ("face-heart-symbolic");
+            image.add_css_class (Granite.STYLE_CLASS_ACCENT);
+            image.add_css_class ("pink");
+
+            var main_label = new Gtk.Label (_("Sponsor Us")) {
+                halign = START,
+                hexpand = true
+            };
+
+            var target_label = new Gtk.Label (null) {
+                halign = START
+            };
+            target_label.add_css_class (Granite.STYLE_CLASS_DIM_LABEL);
+            target_label.add_css_class (Granite.STYLE_CLASS_SMALL_LABEL);
+
+            var level_bar = new Gtk.LevelBar ();
+            level_bar.add_css_class (Granite.STYLE_CLASS_FLAT);
+            level_bar.add_css_class ("pink");
+
+            var details_box = new Gtk.Box (VERTICAL, 0);
+            details_box.append (target_label);
+            details_box.append (level_bar);
+
+            details_revealer = new Gtk.Revealer () {
+                child = details_box,
+                reveal_child = false
+            };
+
+            var link_image = new Gtk.Image.from_icon_name ("adw-external-link-symbolic");
+
+            var grid = new Gtk.Grid () {
+                valign = CENTER
+            };
+            grid.attach (image, 0, 0, 1, 2);
+            grid.attach (main_label, 1, 0);
+            grid.attach (details_revealer, 1, 1);
+            grid.attach (link_image, 2, 0, 2, 2);
+
+            child = grid;
+            add_css_class ("link");
+            get_goal_progress (level_bar, target_label);
+        }
+
+        private void get_goal_progress (Gtk.LevelBar levelbar, Gtk.Label target_label) {
```

Comment:
I'd made this function async and take a GLib.Cancellable to cancel the send_and_read on plug exit.